[DNM] clusterversion: mint 23.2

celiala opened the pull request:
DNM: Using this draft to confirm that tests pass for:
- https://github.com/cockroachdb/cockroach/pull/111634/commits/1de614715d45d9ec2f2233019e189c2ccf3d1685
I plan to ultimately close this in favor https://github.com/cockroachdb/cockroach/pull/111634

celiala committed:
clusterversion: mint 23.2

This mints the release branch as 23.2.
This will be the final backport for the final 23.2 RC.

Release note: none.
Epic: REL-437

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111729)
<!-- Reviewable:end -->

celiala committed:
clusterversion: remove old guidance + check for >= 5 releases ago

Previously, we put this check in place to force the deletion of
old (no longer needed) gates. Since then:
- The team + number of version gates have scaled, so deleting gates
  as part of creating the next development gate is harder to do in
  a single atomic commit. Since 22.2, we've decoupled the deletion
  of old gates from the creation of the next development gate.
- Furthermore, with the introduction of VPrimordial gates, the
  check + error message is now incorrect: the error message says
  to delete 0.0-* gates; but these are VPrimordial gates, which
  should never be deleted.
- This rule assumed that versions older than 5 releases ago are no
  longer active. However, this rule is no longer always true, now
  that we've modified the scheduel for when we advance the minBinary
  (for VersionSkipping).

This commit removes this check, in order to prepare for the creation
of the 24.1 development version gate.

Release note: None.
Epic Link: REL-437.

celiala committed:
build: advance version to beta.1

This needs tp be merged onto release-23.2 right
before selecting the release candidate for 23.2.0-beta.1

Release note: None.
Epic: REL-437.

celiala committed:
clusterversion: set developmentBranch to false

This sets developmentBranch to false for the release branch.
We plan to backport this commit as soon as the branch is cut.

Release note: none.
Epic: REL-437.

celiala commented:
Closing in favor of https://github.com/cockroachdb/cockroach/pull/112347
